e2e/cluster.go

```diff
@@ -180,6 +180,26 @@ func getOrCreateCluster(ctx context.Context, t *testing.T, cluster *armcontainer
 	}
 }
 
+func deleteCluster(ctx context.Context, t *testing.T, cluster *armcontainerservice.ManagedCluster) error {
+	t.Logf("deleting cluster %s in rg %s", *cluster.Name, config.ResourceGroupName)
+	_, err := config.Azure.AKS.Get(ctx, config.ResourceGroupName, *cluster.Name, nil)
+	if err != nil {
+		var azErr *azcore.ResponseError
+		if errors.As(err, &azErr) && azErr.StatusCode == 404 {
+			t.Logf("cluster %s does not exist in rg %s", *cluster.Name, config.ResourceGroupName)
+			return nil
+		}
+		return fmt.Errorf("failed to get cluster %q: %w", *cluster.Name, err)
+	}
+
+	_, err = config.Azure.AKS.BeginDelete(ctx, config.ResourceGroupName, *cluster.Name, nil)
```

Comment:
we might want to actually poll until the delete operation is deleted, from what I'm seeing in the subsequent code paths we'd hit `createNewAKSClusterWithRetry` assuming the provisioning state is `Deleting`, which immediately tries to create the cluster again - we do have retries on the create operation but it's possible deletion won't finish before all those retries are exhausted